modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFolderLocalServiceTest.java

```diff
@@ -0,0 +1,329 @@
+				_uncategorizedObjectFolder.getObjectFolderId()
+			).size();
+
+		objectFolder = _objectFolderLocalService.deleteObjectFolder(
+			objectFolder.getObjectFolderId());
+
+		Assert.assertNull(
+			_objectFolderLocalService.fetchObjectFolder(
+				objectFolder.getObjectFolderId()));
+
+		Assert.assertEquals(
+			0,
+			_objectDefinitionLocalService.getObjectDefinitions(
+				objectFolder.getObjectFolderId()
+			).size());
+
+		Assert.assertEquals(
+			countUncategorizedObjectFolder + 2,
+			_objectDefinitionLocalService.getObjectDefinitions(
+				_uncategorizedObjectFolder.getObjectFolderId()
+			).size());
+	}
+
+	@Test
+	public void testUpdateObjectFolder() throws Exception {
+		ObjectFolder objectFolder1 = _addObjectFolder(
+			false, TestPropsValues.getUser());
+
+		AssertUtils.assertFailure(
+			DuplicateObjectFolderExternalReferenceCodeException.class,
+			StringBundler.concat(
+				"Duplicate object folder with external reference code ",
+				_uncategorizedObjectFolder.getExternalReferenceCode(),
+				" and company ", _uncategorizedObjectFolder.getCompanyId()),
+			() -> _objectFolderLocalService.updateObjectFolder(
+				_uncategorizedObjectFolder.getExternalReferenceCode(),
+				objectFolder1.getObjectFolderId(),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				Collections.emptyList()));
+
+		AssertUtils.assertFailure(
+			ObjectFolderLabelException.class,
+			"Label is null for locale " + LocaleUtil.US.getDisplayName(),
+			() -> _objectFolderLocalService.updateObjectFolder(
+				RandomTestUtil.randomString(),
+				objectFolder1.getObjectFolderId(), null,
+				Collections.emptyList()));
+
+		_objectFolderLocalService.deleteObjectFolder(objectFolder1);
+
+		ObjectFolder objectFolder2 = _addObjectFolder(
+			false, TestPropsValues.getUser());
+
+		String externalReferenceCode = RandomTestUtil.randomString();
+		Map<Locale, String> labelMap = LocalizedMapUtil.getLocalizedMap(
+			RandomTestUtil.randomString());
+
+		objectFolder2 = _objectFolderLocalService.updateObjectFolder(
+			externalReferenceCode, objectFolder2.getObjectFolderId(), labelMap,
+			Collections.emptyList());
+
+		_validateObjectFolder(
+			externalReferenceCode, labelMap, objectFolder2.getName(),
+			objectFolder2);
+
+		_objectFolderLocalService.deleteObjectFolder(objectFolder2);
+	}
+
+	private ObjectFolder _addObjectFolder(
+			boolean addObjectDefinitions, User user)
+		throws Exception {
+
+		ObjectFolder objectFolder = _objectFolderLocalService.addObjectFolder(
+			RandomTestUtil.randomString(), user.getUserId(),
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			RandomTestUtil.randomString());
+
+		if (addObjectDefinitions) {
+			ObjectDefinitionTestUtil.addObjectDefinition(
+				_objectDefinitionLocalService,
+				objectFolder.getObjectFolderId());
+			ObjectDefinitionTestUtil.addObjectDefinition(
+				_objectDefinitionLocalService,
+				objectFolder.getObjectFolderId());
+		}
```

Comment:
I don't think this parameter and its purpose are clear to the developer.
I prefer to make this method only to add an Object Folder. If we need to create an Object Definition, we will create it inside the test. This way, it is clear to the user when creating the Object Definition and how many.

Reply:
Done